website/docs/class-of.md

```diff
@@ -1,75 +1,200 @@
-  → View on sorbet.run
-</a>
+**These errors are correct**, and we can verify them in the Ruby REPL. First,
+let's explain the error on the last line above:
+
+```
+❯ MyClass.singleton_class.ancestors
+=> [#<Class:MyClass>, MyInterface::ClassMethods, #<Class:Object>, T::Private::Methods::MethodHooks, #<Class:BasicObject>, Class, Module, T::Sig, Object, Kernel, BasicObject]
+```
+
+The first two ancestors of the `MyClass` object are itself and
+`MyInterface::ClassMethods`. But notably, `#<Class:MyInterface>` **does not**
+appear in this list, so Sorbet is correct to say that `MyClass` does not have
+type `T.class_of(MyInterface)`. This is because neither `include` nor `extend`
+in Ruby will cause `#<Class:MyInterface>` to appear in any ancestors list.
+
+Next, let's explain the other two errors:
+
+```
+❯ MyInterface.singleton_class.ancestors
+=> [#<Class:MyInterface>, T::Private::MixesInClassMethods, T::Helpers, Module, T::Sig, Object, Kernel, BasicObject]
+```
+
+For the `MyInterface` class object, we see that its only ancestor is itself
+(ignoring those common ancestors). Notably, **none** of the classes in this list
+define either a method called `new` (because `Class` is not there) nor
+`some_class_method` (because `MyInterface::ClassMethods` is not there).
+
+While these errors are correct, **we want to be able to type this code**. There
+are two options:
+
+1.  Use an abstract class instead of an interface.
+
+    Sometimes this is not possible, because the class in question already has a
+    superclass that can't be changed. However, if this is an option to you, it's
+    likely the most straightforward. If we change `MyInterface` to
+    `MyAbstractClass`, all our problems vanish.
+
+2.  Use `T.all(Class, MyInterface::ClassMethods)`.
+
+    This is only a partial solution, but might be good enough.
+
+Specifically, option (2) looks like this:
+
+```ruby
+sig {params(x: T.all(Class, MyInterface::ClassMethods)).void}
+def example3(x)
+  x.new.some_instance_method  # error: `some_instance_method` does not exist
+  x.some_class_method         # ok
+end
+
+example3(MyClass)             # ok
+```
+
+We’re down to only one error now. The error is still technically correct: since
+we’re using `Class` instead of `T.class_of(...)`, Sorbet has no way to know what
+the instance type created by `x.new` will be (it could be anything), so it
+treats the type as `Object`, causing `some_instance_method` to not be found.
+
+But if this tradeoff is okay, both the top-level call site and the call to
+`x.some_class_method` now typecheck successfully.
```

Comment:
```suggestion
We’re down to only one error now. The error is still technically correct: since
we’re using `Class` instead of `T.class_of(...)`, Sorbet has no way to know what
the instance type created by `x.new` will be (it could be anything), so it
treats the type as `Object`, causing `some_instance_method` to not be found.
However, both the top-level call site to `example3` and the call to `x.some_class_method`
now typecheck successfully. In cases where we don't actually need to use instance methods
from `MyInterface`, this may be an acceptable workaround.
```